jupyter-js-widgets/src/widget_string.ts

```diff
@@ -385,3 +385,121 @@ class TextView extends LabeledDOMWidgetView {
 
     textbox: HTMLInputElement;
 }
+
+export
+class PasswordModel extends StringModel {
+    defaults() {
+        return _.extend(super.defaults(), {
+            _view_name: 'PasswordView',
+            _model_name: 'PasswordModel'
+        });
+    }
+}
+
+export
+class PasswordView extends LabeledDOMWidgetView {
```

Comment:
In fact, why don't we change `TextView` to have a new protected attribute (at the bottom of the class), the `inputType`, which for `TextView` is `'text'` and is used in the `render()` method, but the `PasswordView` subclass just sets this to `'password'`. Then the PasswordView just becomes something like:

```typescript
export
class PasswordView extends TextView {
    inputType: string = 'password';
}
```
or something like that.